pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -437,6 +436,10 @@ func (r *RootSyncReconciler) SetupWithManager(mgr controllerruntime.Manager, wat
 			handler.EnqueueRequestsFromMapFunc(r.mapObjectToRootSync),
 			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})).
 		Watches(&source.Kind{Type: &rbacv1.ClusterRoleBinding{}},
+			handler.EnqueueRequestsFromMapFunc(r.mapObjectToRootSync),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})).
+		// TODO: is it possible to watch with a label filter?
```

Comment:
This is supported https://github.com/kubernetes-sigs/controller-runtime/issues/244

Reply:
Dug into this a bit and it looks like the implementation you linked is a setting at the level of the controller manager (`SelectorsByObject`). Since we use a shared controller manager for both RootSync and RepoSync reconcilers, I don't think we could filter all RoleBindings by a single label selector (based on the current labels).

This could potentially work if we added a new label that was applied to all managed objects (RoleBindings in this case, but could also apply to other objects) that is the same for both RootSync and RepoSync

Reply:
Generally when using the controller-manager you have to assume your watches are being shared, so you can't filter them without filtering them for all controllers (e.g. [SelectorsByObject](https://github.com/kubernetes-sigs/controller-runtime/pull/1435)). There was discussion of making these dynamic based on watch selectors here, but it was never implemented due to caching complexities.

The general workaround is to use WithPredicates instead, which allows skipping reconciling of events for a specific controller unless they match the predicate.

Reply:
There's a builder.LabelSelectorPredicate, if you want.

Reply:
This could be added as a future optimization.